### PR TITLE
Refactor cgroup tests and fix node creation

### PR DESCRIPTION
**Pull request by mike0042**
#### Bug/feature Description
* Fix vnode creation and refactor several issues with cgroup tests

#### Affected Platform(s)
* Linux

#### Cause / Analysis / Design
* Nodes were not being properly configured on systems where the network name and hostname didn't match.
* Hostnames were being stored in "mom*" variables and mom dictionaries were being stored in "host*" variables

#### Solution Description
* Use the correct node names when creating vnodes
* Refactor variable names

#### Testing logs/output
* [ptl.log](https://github.com/PBSPro/pbspro/files/2018386/ptl.log)

#### Checklist:

- [X] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [X] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [X] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [X] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [X] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_cgroups_hook.py [hunk 1]
```diff
@@ -661,23 +675,34 @@ def get_paths(self):
                     paths['devices'] = paths[subsys]
         return paths
 
-    def is_dir(self, cpath, mom):
+    def is_dir(self, cpath, host):
```
[the remark this answers is not in the document]
Comment (mike0042) [reply test/tests/functional/pbs_cgroups_hook.py]: Done.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 2]
```diff
@@ -1002,10 +1039,10 @@ def test_cgroup_periodic_update(self):
             self.assertGreater(30000, usage)
         # Allow some time to pass for values to be updated
         self.logger.info('Waiting for periodic hook to update usage data.')
-        time.sleep(5)
+        time.sleep(15)
```
[the remark this answers is not in the document]
Comment (mike0042) [reply test/tests/functional/pbs_cgroups_hook.py]: Done.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 3]
```diff
@@ -418,7 +418,7 @@ def setUp(self):
             "exclude_hosts"   : [],
             "exclude_vntypes" : [],
             "soft_limit"      : false,
-            "default"         : "256MB",
+            "default"         : "96MB",
```
Comment (bhroam) [inline test/tests/functional/pbs_cgroups_hook.py]: Why the change between 96mb and 256mb?  Both seem rather low for any program these days.

**Comment by mike0042** [comment]
Closing and reopening to restart codacy check.

File: src/hooks/cgroups/pbs_cgroups.PY [hunk 4]
```diff
@@ -1809,11 +1809,18 @@ class NodeConfig(object):
...
         pbs.logmsg(pbs.EVENT_DEBUG4, '%s: numa nodes: %s' %
                    (caller_name(), self.numa_nodes))
+        vnode_name = self.hostname
+        # In some cases the hostname and vnode name do not match
+        if vnode_name not in vnode_list:
+            vnkeys = vnode_list.keys()
+            if len(vnkeys) == 1:
+                vnode_name = vnkeys[0]
+            else:
+                raise ProcessingError('Could not identify local vnode')
```
Comment (bhroam) [inline src/hooks/cgroups/pbs_cgroups.PY]: I don't understand why the length of the vnode_list has to be 1?  Why is it that if the length of vnode_list is not one, we can't identify the local vnode?

File: src/hooks/cgroups/pbs_cgroups.PY [hunk 5]
```diff
@@ -219,7 +219,7 @@ def convert_size(value, units='b'):
         return str(val) + new
     except KeyboardInterrupt:
         raise
-    except:
+    except Exception:
```
[the remark this answers is not in the document]
Comment (mike0042) [reply src/hooks/cgroups/pbs_cgroups.PY]: We still need to handle KeyboardInterrupt independent of other exceptions so that when the timer pops we don't ignore it and keep running.

File: src/hooks/cgroups/pbs_cgroups.PY (hunk 4 above)
Comment (mike0042) [reply src/hooks/cgroups/pbs_cgroups.PY]: If there were more than one in the list, which one would we choose? There's really no way to know which is the correct one. The create_vnodes() method is only called by the exechost_startup handler, so there *should* only be one vnode key in the dictionary. If there happens to be more than one, all we can do is raise an error.

File: src/hooks/cgroups/pbs_cgroups.PY (hunk 5 above)
[the remark this answers is not in the document]
Comment (bhroam) [reply src/hooks/cgroups/pbs_cgroups.PY]: I don't think you do any more.  KeyboardInterrupt inherits BaseException, not Exception.  You are now doing an except Exception instead of a blanket except.  A KeyboardInterrupt will be unhandled.

File: src/hooks/cgroups/pbs_cgroups.PY (hunk 4 above)
Comment (bhroam) [reply src/hooks/cgroups/pbs_cgroups.PY]: Is this a point before you add more to the dictionary?  If there is only ever going to be one, why make it a dictionary?

Comment (mike0042) [reply src/hooks/cgroups/pbs_cgroups.PY]: That's how it's passed to the hook, and that's how it gets passed back. It's part of the hooks infrastructure. Am I missing your point?

Comment (bhroam) [reply src/hooks/cgroups/pbs_cgroups.PY]: Oh, I understand.  It's coming from the hook infrastructure before we create vnodes.  If there is more than one vnode, we have a problem.  Won't this fail if someone set up a vnodedef file?  Then we'll have more than one vnode.

File: src/hooks/cgroups/pbs_cgroups.PY (hunk 5 above)
[the remark this answers is not in the document]
Comment (mike0042) [reply src/hooks/cgroups/pbs_cgroups.PY]: You lost me. In the event of a KeyboardInterrupt it will throw an error that will eventually be caught somewhere up the stack. In the event of any other exception, the function returns None.

File: src/hooks/cgroups/pbs_cgroups.PY (hunk 4 above)
Comment (mike0042) [reply src/hooks/cgroups/pbs_cgroups.PY]: So *IF* we're on a system with a wonky hostname configuration *AND* the node has a vnodedef file with multiple vnodes, it will raise a ProcessingError. It's a "best effort" scenario.

File: src/hooks/cgroups/pbs_cgroups.PY (hunk 5 above)
[the remark this answers is not in the document]
Comment (vstumpf) [reply src/hooks/cgroups/pbs_cgroups.PY]: `except Exception` does not catch every exception.
As seen here: https://docs.python.org/2/library/exceptions.html#exception-hierarchy
```
BaseException
 +-- SystemExit
 +-- KeyboardInterrupt
 +-- GeneratorExit
 +-- Exception
```
Everything in `Exception` will be caught, and that excludes KeyboardInterrupts.

[the remark this answers is not in the document]
Comment (mike0042) [reply src/hooks/cgroups/pbs_cgroups.PY]: Point take. So is the request to remove the except for KeyboardInterrupt now that we're no longer using blanket excepts?

[the remark this answers is not in the document]
Comment (vstumpf) [reply src/hooks/cgroups/pbs_cgroups.PY]: Exactly.

File: src/hooks/cgroups/pbs_cgroups.PY (hunk 4 above)
Comment (bhroam) [reply src/hooks/cgroups/pbs_cgroups.PY]: As long as we're not making the cgroups hook and vnode def files incompatible, I'm happy with this.

**Comment by bhroam** [comment]
I merged in the PR

**Comment by vstumpf** [comment]
Thanks Bhroam!
